Design/Muon/ADS_data_structure.md

```diff
@@ -0,0 +1,285 @@
+
+EMU0001235/
+    EMU0001235; Pair long; 1+2; #1
+    EMU0001235; Pair long; 1+2; #1_Raw
+    EMU0001235; Pair long; 1+2; #1_unNorm
+    EMU0001235; Pair long; 1+2; #1_Raw_UnNorm
+
+    EMU0001235; Group fwd; 1; #1
+    EMU0001235; Group fwd; 1; #1_Raw
+    EMU0001235; Group fwd; 1; #1_unNorm
+    EMU0001235; Group fwd; 1; #1_Raw_UnNorm
+
+    ...
+
+MuonAnalysisTFNormalizations
+
+MuonAnalysisGrouped/
+    MuonAnalysis_Grouped_1
+```
+
+## Issues with the current structure
+
+Below are some issues I think are making this structure either hard to understand or hard to use from a user's perspective;
+
+- Generally speaking, there tends to be a huge proliferation of workspaces within each folder, many of which the user simply does not need to see or interact with (e.g. the `unNorm` which were only added a few months ago).
+
+- Different combinations of periods are held within the run folder. For example I can redo the same analysis with periods 1 and 2 added together, or periods 1 and 2 subtracted. These are fundamentally different analyses and should be kept separate.
+
+- Grouping and pairing data are held within the same run folder, despite these being conceptually different objects.
+
+- The `MuonAnalysis` workspace enforces that we can only deal with one file at a time; there is no reason to impose this and in fact it would be better to be able to deal with several files simultaneously.
+
+- The `MuonAnalysis` being different kinds of objects depending on whether the data is multi-period or not is unneccessary and confusing; why not simply have a group in all cases, with just one workspace for single period data?
+
+# Proposed Modifications
+
+I propose to keep the data structure and naming *essentially* the same, with a few minor modifications which act to make the structure more readable, cleaner, and gather data together which is likely to be accessed at once, and to hide data that the user does not need to interact with.
+
+Modifications to the current structure are : 
+
+- Another layer of folders within each run folder; one for groups, pairs and group-asymmetries.
+
+- A raw data folder in each run folder, to prevent re-loading. This does away with the odd `MuonAnalysis` workspace or workspace group. Single period data is then represented by just one workspace in the raw folder, this suggests thinking about single period data as a limiting case of multi-period data which makes much cleaner analyses.
+
+- Hide the extra workspaces (raw, unNorm, raw_unNorm). The user can still access the data if they want to.
+
+- Each set of period combinations has its own folder; this way we don't mix different periods combinations in the same folder. There is never a scenario where this is a good idea as the names are so similar that on a quick glance they look identical and it is so easy to accidentally select the wrong one.
+
+
+```python
+EMU0001234/
+
+    EMU0001234_raw_data/
```

Comment:
As this is just cached data (like the other cached workspaces then why not have a single group workspace for all of those here? including those you were thinking about hiding?
Most people will never look inside.  

How about EMU001234_data/
                            EMU001234_raw_data/
                            EMU001234_cached_data/ (for the other workspaces you were thinking about hiding?

Reply:
Yes, this would be better and would remove the need for hiding them altogether.